plugin/fallback/setup.go

```diff
@@ -0,0 +1,53 @@
+package fallback
+
+import (
+	"fmt"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/proxy"
+	"strings"
+
+	"github.com/mholt/caddy"
+	"github.com/miekg/dns"
+)
+
+func init() {
+	caddy.RegisterPlugin("fallback", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
+	t := dnsserver.GetConfig(c).Handler("trace")
+	f := &Fallback{trace: t, rules: make(map[int]proxy.Upstream)}
+
+	for c.Next() {
+		var rcode string
+		if !c.Dispenser.Args(&rcode) {
+			return c.ArgErr()
+		}
+
+		fmt.Printf("Found rcode %s\n", rcode)
+		rc, ok := dns.StringToRcode[strings.ToUpper(rcode)]
+		if !ok {
+			return fmt.Errorf("%s is not a valid rcode", rcode)
+		}
+
+		fmt.Printf("Maps to rcode %d\n", rc)
+
+		u, err := proxy.NewStaticUpstream(&c.Dispenser)
+		if err != nil {
+			return plugin.Error("fallback", err)
+		}
+
+		f.rules[rc] = u
```

Comment:
Hum.. for what I understand, this setup is called once per word 'fallback' found in the config file.
Therefore, the way the syntax of this stanza is designed, you can register only ONE rcode per 'fallback'. 
in other word this map named rules could be replaced by a couple (int, Upstream).
Am  I correct ?

Reply:
yes, one `rcode` per `fallback` directive. See the comment above - I changed the syntax from the original, so it would be easy to re-use the `proxy` code.

There is only *one* fallback plugin, not *one per directive*. So, this one plugin processes *all* `fallback` directives, and so they end up in this map.

Reply:
maybe I misread the setup call ...
Ok. I double checked when is called the setup .. and it says **once per key for each server block**.
Very good. - exactly what is needed :)

/```
/ SetupFunc is used to set up a plugin, or in other words,
// execute a directive. It will be called once per key for
// each server block it appears in.
type SetupFunc func(c *Controller) error
```